pyresample/spherical.py

```diff
@@ -42,14 +98,16 @@ class SCoordinate(object):
     """
 
     def __init__(self, lon, lat):
-        if np.isfinite(lon):
-            self.lon = float(_unwrap_radians(lon))
-        else:
-            self.lon = float(lon)
+        lon = np.asarray(lon, dtype=np.float64)
```

Comment:
It looks like the check for finite values is gone, why is this not needed anymore?